### DISCUSSION TITLE
How to useFetcher with always-zero reloadings

### DISCUSSION THREAD

**Discussion by meglio** [remix-run/remix#4438] [Q&A]
I use `useFetcher()` to save website interface settings as the user updates them  - e.g. shows/hides the global sidebar menu, collapes/expands sections in it, etc.

I want zero reloads after such a save. How can I achieve it with `useFetcher()`?

Reloads are not needed because it is a one-way information flow: I want interface settings saved as they are changed, but there is absolutely no need to reload routes after the save - because the UI will already be in the correct state. You can think of this as if the "ui save" action follows real UI and saves the updated settings in the background, so that when the user reloads the page the next time, the UI is how he left it.

Using `unstable_shouldReload()` looks like an overkill because I need to embed it in all routes and tell to not reload on the save-ui-settings action? Or am I misunderstanding how `unstable_shouldReload` should work and where to declare it?

P.S. As a temporary workaround, I am using loader instead of action to save the UI settings. But it breaks the convention of using loaders to load data and actions to edit data, so this approach does not feel like the right solution.

**Comment by sergiodxa** [comment]
The only solution is to use `unstable_shouldReload` in every route since I suppose that UI is global.

**Reply by dusty** [reply]
What if you just use `fetch` instead of `useFetcher`?

**Reply by meglio** [reply]
"The only solution is to use unstable_shouldReload" - does not look like a solution, sorry. I would not even say it is useful as a workaround, because you can't keep adding `unstable_shouldReload` to every new route as the project grows just to make one simple thing work. The workaround I ended up with (and described above) is much more elegant according to practical reasons, but is still a workaround and does not feel right.

**Reply by machour** [reply]
@meglio I get your pain here. Have you considered @dusty solution? Seems like it would do the trick and still allow you to POST to an `action`

**Reply by meglio** [reply]
@machour ok a few questions:

1) by this, do I bypass the out-of-the-box client-server communication channel that Remix maintains?

2) is `fetch` polyfilled by Remix in the browser environment the same way it is in the server one?

3) what features will I lose by using `fetch` instaed of `useFetcher` in this particular scenario, anything important that pops up?

Many thanks!

**Reply by machour** [reply]
1. Not really. You'll just be using one particular route (the one with the action) link an API endpoint instead of a regular route. No biggies.
2. I'm not sure about this one (still a JS noob), maybe @sergiodxa can help answer this, but I think `fetch` support is okay (https://caniuse.com/fetch)
3. You'll be loosing the feature that currently annoys you: Remix won't automatically reload routes for you. Nothing else that pops up in the case you're describing, unless maybe auth? (how do you identify the user performing the change?)

**Reply by sergiodxa** [reply]
Fetch is not polyfilled by Remix on the browser, neither is on the server if the runtime supports fetch, so only in Node.

**Reply by dusty** [reply]
1.  Not sure what that means
2. You could use something like axios if you are worried about fetch support (ie) or use a polyfill (https://github.com/github/fetch#installation).
3. useFetcher maintains state (state, submission, type, data, etc..), but it sounds like your use case doesn't need any of that.

**Reply by sergiodxa** [reply]
For Fetch API support, Remix only send JS code with `<script type="module">`, which is ignored by old browser, and any browser that supports module JS files already supports Fetch API, so if your client-side JS loads you have support for Fetch API on that browser.

This also means IE don't load Remix client-side JS code. The only way to use client-side JS code in IE when using Remix is to manually add a `<script>` tag pointing to a JS file inside the public folder, so completely outside Remix.
